### DISCUSSION TITLE
Core event handlers on window instead of container?

### DISCUSSION THREAD

**Discussion by jcuypers** [cytoscape/cytoscape.js#3106] [Q&A] [answered]
Hi,   

I was wondering if anybody knew the reason why some of the event handlers for the core cytoscape js functionality are attached to the window object instead of the cytoscape container.  Specifically for mouseup/down/move events.  Is there a technical need for it?  

Thanks, 
J.

**Comment by maxkfranz** [comment, marked as answer]
They're necessary for users to have gestures behave in the ways that they'd expect.

**Comment by jcuypers** [comment]
Hi @maxkfranz,  thanks for the feedback.  

The nature of my question is that I'm trying to implement a popout functionality (window open / react portal) functionality with a cytoscapejs  / react cytoscape.

I have reached the point where most things work except the listeners (if I move them to the container it works).

Based on your info, I think my approach should be that whenever I popout, I should clear the bindings and reapply them for it to work correctly.

Is this a functionality that you would to see as a PR?

Thanks,
J.

**Reply by maxkfranz** [reply]
Exposing external control of the internal event handling would introduce complexity and opportunities for user error.  Best for to apps to manage app-level event handling at the app level.

Perhaps Plotly would be open to a PR re. easier use of `createPortal` in [`plotly/react-cytoscapejs`](https://github.com/plotly/react-cytoscapejs).
